### PR TITLE
🚨 Fix remaning GCC warnings

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/scrollrt.cpp [hunk 1]
```diff
@@ -497,8 +497,8 @@ static void DrawObject(const CelOutputBuffer &out, int x, int y, int ox, int oy,
 		return;
 	}
 
-	int nCel = object[bv]._oAnimFrame;
-	auto frames = SDL_SwapLE32(reinterpret_cast<const std::uint32_t *>(pCelBuff)[0]);
+	uint32_t nCel = object[bv]._oAnimFrame;
+	auto frames = SDL_SwapLE32(reinterpret_cast<const uint32_t *>(pCelBuff)[0]);
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/scrollrt.cpp:501]: It's the very start of a file loaded in memory, so I would say it is.

[the remark this answers is not in the document]
Comment (glebm) [reply Source/scrollrt.cpp:501]: > It's the very start of a file loaded in memory, so I would say it is.

Don't we allocate the buffer for it as `new uint8_t[]`? If so, it is not guaranteed to be.

[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/scrollrt.cpp:501]: Afaik LoadLE32 can operate on std::byte so in that case we don't need the cast at all.

```suggestion
	auto frames = SDL_SwapLE32(pCelBuff);
```

[the remark this answers is not in the document]
Comment (glebm) [reply Source/scrollrt.cpp:501]: Nice :+1:

**Pull request merged**